lib/hacluster.pm

```diff
@@ -974,49 +976,64 @@ Calculates start time delay after node is fenced.
 Prevents cluster failure if fenced node restarts too quickly.
 Delay time is used either if specified in sbd config variable "SBD_DELAY_START"
 or calculated:
-"corosync_token + corosync_consensus + SBD_WATCHDOG_TIMEOUT * 2"
+"corosync token timeout + consensus timeout + pcmk_delay_max + msgwait"
 Variables 'corosync_token' and 'corosync_consensus' are converted to seconds.
+For diskless SBD pcmk_delay_max is set to static 30s.
+
+%sbd_parameters = {
+    'corosync_token' => <runtime.config.totem.token>,
+    'corosync_consensus' => <runtime.config.totem.consensus>,
+    'sbd_watchdog_timeout' => <SBD_WATCHDOG_TIMEOUT>,
+    'sbd_delay_start' => <SBD_DELAY_START>,
+    'pcmk_delay_max' => <pcmk_delay_max>
+}
+
 =cut
 
 sub calculate_sbd_start_delay {
-    my %params;
+    my ($sbd_parameters) = @_;
+    my %params = $sbd_parameters ? %$sbd_parameters : (
+        'corosync_token' => script_output("corosync-cmapctl | awk -F \" = \" '/runtime.config.totem.token\\s/ {print \$2\/1000}'"),
+        'corosync_consensus' => script_output("corosync-cmapctl | awk -F \" = \" '/runtime.config.totem.consensus\\s/ {print \$2\/1000}'"),
+        'sbd_watchdog_timeout' => script_output("grep -oP '(?<=^SBD_WATCHDOG_TIMEOUT=)[[:digit:]]+' /etc/sysconfig/sbd"),
+        'sbd_delay_start' => script_output("grep -oP '(?<=^SBD_DELAY_START=)([[:digit:]]+|yes|no)+' /etc/sysconfig/sbd"),
+        'pcmk_delay_max' => get_var('USE_DISKLESS_SBD') ? 30 :
+          script_output("crm resource param stonith-sbd show pcmk_delay_max| sed 's/[^0-9]*//g'")
+    );
+
     my $default_wait = 35 * get_var('TIMEOUT_SCALE', 1);
 
-    %params = (
-        'corosync_token' => script_output("corosync-cmapctl | awk -F \" = \" '/config.totem.token\\s/ {print \$2}'"),
-        'corosync_consensus' => script_output("corosync-cmapctl | awk -F \" = \" '/totem.consensus\\s/ {print \$2}'"),
-        'sbd_watchdog_timeout' => script_output("awk -F \"=\" '/SBD_WATCHDOG_TIMEOUT/ {print \$2}' /etc/sysconfig/sbd"),
-        'sbd_delay_start' => script_output("awk -F \"=\" '/SBD_DELAY_START/ {print \$2}' /etc/sysconfig/sbd")
-    );
+    record_info('SBD Params', Dumper(\%params));
 
     # if delay is false return 0sec wait
-    if ($params{'sbd_delay_start'} == 'no' || $params{'sbd_delay_start'} == 0) {
-        record_info("SBD start delay", "SBD delay disabled in config file");
+    if (grep /^$params{'sbd_delay_start'}$/, qw(no 0)) {
```

Comment:
Hmm... good point. I think perl code will poop itself since $params{'sbd_delay_start'} does not exist. 
I am thinking about putting the undefined value into the  'yes' branch and if the value is not provided, delay will be calculated from parameters. 
https://github.com/lpalovsky/os-autoinst-distri-opensuse/blob/7412a5ff3fa15bc5f07ff4aa577ddc533af0edb3/lib/hacluster.pm#L1015
I guess that would apply as well in case if sbd config file does not include the parameter as well. 
And this means I need to update unit test too.